### ISSUE TITLE
Setup real cron

### ISSUE THREAD

**Issue by fgilio** [wplib/wplib-box#166]
DISABLE_WP_CRON by default + Setup real cron

So, by default add this in `wp-config.php` 
```php
define('DISABLE_WP_CRON', 'true');
```

And automatically setup an actual cron job in the VM. This would have to be done for each domain, taking into account the future update of wplib-box

Some resources about this: 
- https://hondo.wtf/2015/01/11/performant-wordpress-disabling-wp-cron 
- https://www.lucasrolff.com/wordpress/why-wp-cron-sucks/ 
- http://stickystatic.com/tech/disable-wordpress-cron-jobs 
- http://www.inmotionhosting.com/support/website/wordpress/disabling-the-wp-cronphp-in-wordpress 
- http://support.hostgator.com/articles/specialized-help/technical/wordpress/how-to-replace-wordpress-cron-with-a-real-cron-job

:D

**Comment by mikeschinkel** [comment]
@fgilio Thanks for the suggestion.

So here are my current thoughts.  We should implement a site configuration option that allows the user to choose wp-cron vs. real cron but wp-cron would be the default so that people who do not even know about the cron issue do not build sites locally and have no problem with wp-cron and then launch a site and find out there are lots of issues.  OTOH those who disable cron are more likely to know about it and also do the same on the server.

Does that not sound like a good plan?

That said, what specific things are you running into where this becomes an issue or interest such that you would request it?  Not challenging your need, just trying to understand it the best that I can.

**Comment by fgilio** [comment]
Hi!
It sounds good. Totally understand why this would be optional, and also agree.

But I think, and correct me if I'm wrong, the target audience for wplib-box is most likely understand what a cron is and prefer to have a real one running. Maybe this is changing in the near future... you know best.


Regarding the reason. I work on some big websites that need both and accurate cron system and also one that won't hammer the servers if the number of users grow. Also, I'd like to have a local development environment as close as possible to the production one.

**Comment by mikeschinkel** [comment]
> _But I think, and correct me if I'm wrong, the target audience for wplib-box is most likely understand what a cron is and prefer to have a real one running._

We currently believe that in order to be able to have a business that can supported us we will need to address the needs of people who do not understand cron nor understand why they would want one.  We think the target market of people who understand and want it is just too small to support us financially, and also we think they tend to be the kind who are more likely to think _"I can just build that myself!"_ so would be some of the hardest to get to pay for anything.

How we intend to be different from MAMP, Desktop Server, et. al. is we want to also provide a solution that does not limit really serious developers in the way that those solutions are limited mostly to single developers.

So yes, we want and need to address your needs but we also want to provide defaults that follow [the principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) for people who are not as highly skilled as developers like yourself.

P.S. To be clear we intend to deliver solutions that -- for individual use at least -- skilled developers would not need to pay for anything.  Our paid offerings would be targeting those who would rather spend money than be burdened to level up their skills when they either don't have time or inclination to do so.

**Comment by fgilio** [comment]
Of course! I understand, and that's what I was referring to with the "you know best" part.

PS: I've just learnt something new with Principle of least astonishment :D thanks!

**Comment by clubdeuce** [comment]
@mikeschinkel Where do we stand on this? Will we be implementing this, or may I close this ticket?

**Comment by mikeschinkel** [comment]
@MickMakes What do you think about this?  A container for a cron that can optionally be turned on using information in `project.json`?
